src/NLog/LogFactory.cs

```diff
@@ -1153,7 +1151,16 @@ private static IEnumerable<string> GetDefaultCandidateConfigFilePaths()
 #endif
         }
 
-        private Logger GetLogger(LoggerCacheKey cacheKey)
+        internal Logger GetContextLogger(string loggerName, Type concreteType)
```

Comment:
Could you explain this a bit? I see we need a new logger with unique name, isn't? 

Could we rename this to CreateNewUniqueLogger? (or something like that)

Reply:
Sure